allensdk/brain_observatory/behavior/data_files/neuropil_corrected_file.py

```diff
@@ -0,0 +1,71 @@
+import json
+from typing import Dict, Union
+from pathlib import Path
+
+import h5py
+from cachetools import cached, LRUCache
+from cachetools.keys import hashkey
+
+import pandas as pd
+
+from allensdk.internal.api import PostgresQueryMixin
+from allensdk.internal.core import DataFile
+
+
+def from_json_cache_key(cls, dict_repr: dict):
+    return hashkey(json.dumps(dict_repr))
+
+
+def from_lims_cache_key(cls, db, ophys_experiment_id: int):
+    return hashkey(ophys_experiment_id)
+
+
+class NeuropilCorrectedFile(DataFile):
+    """A DataFile which contains methods for accessing and loading
+    neuropil traces, neuropil corrected traces.
+    """
+
+    def __init__(self, filepath: Union[str, Path]):
+        super().__init__(filepath=filepath)
+
+    @classmethod
+    @cached(cache=LRUCache(maxsize=10), key=from_json_cache_key)
+    def from_json(cls, dict_repr: dict) -> "NeuropilCorrectedFile":
+        filepath = dict_repr["neuropil_corrected_file"]
+        return cls(filepath=filepath)
+
+    def to_json(self) -> Dict[str, str]:
+        return {"neuropil_corrected_file": str(self.filepath)}
+
+    @classmethod
+    @cached(cache=LRUCache(maxsize=10), key=from_lims_cache_key)
+    def from_lims(
+        cls, db: PostgresQueryMixin,
+        ophys_experiment_id: Union[int, str]
+    ) -> "NeuropilCorrectedFile":
+        query = """
+                SELECT wkf.storage_directory || wkf.filename AS neuropil_corrected_file
+                FROM ophys_experiments oe
+                JOIN well_known_files wkf ON wkf.attachable_id = oe.id
+                JOIN well_known_file_types wkft
+                ON wkft.id = wkf.well_known_file_type_id
+                WHERE wkf.attachable_type = 'OphysExperiment'
+                AND wkft.name = 'NeuropilCorrection'
+                AND oe.id = {};
+                """.format(ophys_experiment_id)
+        filepath = db.fetchone(query, strict=True)
+        return cls(filepath=filepath)
+
+    @staticmethod
+    def load_data(filepath: Union[str, Path], **kwargs) -> pd.DataFrame:
+        with h5py.File(filepath, 'r') as in_file:
+            traces = in_file['FC'][()]
+            roi_id = in_file['roi_names'][()]
+            r_values = in_file['r'][()]
+            rmse_values = in_file['RMSE'][()]
+            idx = pd.Index(roi_id, name='cell_roi_id').astype('int64')
+            return pd.DataFrame(
+                {
+                    'corrected_fluorescence': list(traces),
```

Comment:
Go ahead and add the `r` and `RMSE` values here to this method. When I chatted about this conceptually with, we were conceptually fine with a `data_file` object being the interface that retrieves and loads the data from a given file. That includes all data in that file that we will use downstream.

Reply:
This will fail the tests that compare the cell_specimen object stored to nwb and reloaded from nwb and the cell_specimen object loaded from data files.

Reply:
But you don't need to copy the per my above comment on `corrected_fluoresce_traces.py` correct?

Reply:
Yeah, I was thinking about this. I think I can just convert the series to a dataframe to pass the validation

Reply:
yep it worked